korma/src/commonTest/kotlin/com/soywiz/korma/geom/PointTest.kt

```diff
@@ -14,4 +14,57 @@ class PointTest {
 
         assertEquals("(0, 5)", Point.fromPolar(0.degrees, 10.0).setToPolar(90.degrees, 5.0).toString())
     }
+
+    @Test
+    fun testAngle() {
+        assertEquals(0.degrees, Point().setToZero().angle)
+        assertEquals(45.degrees, Point().setToOne().angle)
+        assertEquals(0.degrees, Point().setToRight().angle)
+        assertEquals(90.degrees, Point().setToDown().angle)
+        assertEquals(180.degrees, Point().setToLeft().angle)
+        assertEquals(270.degrees, Point().setToUp().angle)
+    }
+
+    @Test
+    fun testChangeLength_WITH_different_types_MUST_keep_angle_degrees() {
+        val point = Point(5, 5)
+        assertEquals(45.degrees, point.angle)
+
+        // Double
+        point.changeLength(10.0)
+        assertEquals(10.0, point.length)
+        assertEquals(45.degrees, point.angle)
+
+        // Float
+        point.changeLength(20f)
+        assertEquals(20.0, point.length)
+        assertEquals(45.degrees, point.angle)
+
+        // Int
+        point.changeLength(40)
+        assertEquals(40.0, point.length)
+        assertEquals(45.degrees, point.angle)
+    }
+
+    @Test
+    fun testChangeLength_ISNOT_accurate_in_all_scenarios() {
+        val point = Point(5, 5)
+        assertEquals(45.degrees, point.angle)
+
+        point.changeLength(30.0)
+        assertEquals(29.999999999999996, point.length)
```

Comment:
well in any case  you can use the third parameter in `assertEquals` to compare floating point values to set the epsilon value so you can use assertEquals(30.0, point.length, 0.01)

Reply:
Good idea and I believe it is a good solution for this scenario:
https://github.com/korlibs/korge-next/pull/712/files#diff-8413bc61de5cb0946163fc468a74c95a953727168a2a228a65b02c9fa7f47c8cR55